src/test/java/org/isf/menu/Tests.java

```diff
@@ -321,12 +323,52 @@ void testIoUpdateUserGroup() throws Exception {
 		UserGroup foundUserGroup = userGroupIoOperationRepository.findById(code).orElse(null);
 		assertThat(foundUserGroup).isNotNull();
 		foundUserGroup.setDesc("Update");
-		assertThat(menuIoOperation.updateUserGroup(foundUserGroup)).isTrue();
+		assertThat(menuIoOperation.updateUserGroup(foundUserGroup).getCode()).isEqualTo(foundUserGroup.getCode());
 		UserGroup updatedUserGroup = userGroupIoOperationRepository.findById(code).orElse(null);
 		assertThat(updatedUserGroup).isNotNull();
 		assertThat(updatedUserGroup.getDesc()).isEqualTo("Update");
 	}
 
+	@Test
+	void testMgrSoftDeleteUser() throws Exception {
+		String userName = setupTestUser(false);
+		User user = userBrowsingManager.getUserByName(userName);
+		assertThat(user).isNotNull();
+		assertThat(user.isDeleted()).isFalse();
+
+		user.setDeleted(true);
+		User foundUser = userIoOperationRepository.findByUserName(userName);
+		assertThat(foundUser).isNotNull();
```

Comment:
how do this work????   you are setting the `isDeleted` field on the local object `user`.   It is never persisted to the database.   So when you `findByUserName()` shouldn't the `foundUser` be equal to the original `user` (that is, not deleted)?

And why in line 335 do you use `getUserByName()` and in line 340 you are using `findByUserName()`?  What about `getUserByNameAndIsDeleted()`?

Reply:
I've passed several time in debuging that. You can see that tests are passing. When I update  user `user.setDeleted(true)`, the change applies automatically in the db. Maybe it's related with hibernate configuration.
getUserByName will retrieve non soft deleted user. repository.findByUsername will retrieve user whether soft deleted or no. getUserByNameAndIsDelete with retrieve only soft deleted user.

Reply:
Found that JPA [default for flush mode is AUTO](https://www.baeldung.com/spring-jpa-flush#:~:text=The%20two%20main%20flush%20modes,up%2Dto%2Ddate%20data.) means that JPA flushes the persistence context automatically.
If we set `entityManager.setFlushMode(FlushModeType.COMMIT);` in `OHCoreTestCase`, the test fails, along with 118 similar tests...
Doing so we could enforce the codebase, although I'm not sure would be the best practise... wdyt?

Reply:
I'm not proud to have fixed tests that way too. We could address that issue in a refactoring task. Wdyt ?

Reply:
If we decide it worth, for sure